cmd/kubeadm/app/master/pki.go

```diff
@@ -158,7 +158,7 @@ func CreatePKIAssets(cfg *kubeadmapi.MasterConfiguration) (*rsa.PrivateKey, *x50
 	}
 	altNames.DNSNames = append(altNames.DNSNames, cfg.API.ExternalDNSNames...)
 
-	pkiPath := path.Join(kubeadmapi.GlobalEnvParams.HostPKIPath)
+	pkiPath := kubeadmapi.GlobalEnvParams.HostPKIPath
```

Comment:
This isn't equivalent.  `path.Join` will also clean the path.  If you want to remove the Join, you should replace it with `path.Clean`.  See: https://play.golang.org/p/jLUuVGBBSy